articles/2016-10-12-functor-adts.txt

```diff
@@ -1,37 +1,38 @@
 package[value="articles"]{}
 
-post[id=functor-adts,date=2016-10-12,description="Using contravariance, type parameters and type members to implement a functor in ADTs",Implementing functors in ADTs] {
+post[id=functor-adts,date=2016-10-12,description="Review of Scala language features for call site type polymorphism",Call site type polymorphism] {
   section[Motivation] {
-    It is a common problem to define url[https://en.wikipedia.org/wiki/Algebraic_data_type]{Algebraic Data Types} (ADTs). To prevent duplication, shared functionality should be ideally kept in base code{trait}s. Oftentimes, a problem arises that in the base code{trait} we do not know the child type the function is called on.
+    It is a common problem to define class hierarchies. To prevent duplication, shared functionality should be ideally kept in base code{trait}s. Oftentimes, a problem arises that in the base code{trait} we do not know the child type the function is called on:
```

Comment:
We do not know the type the child method returns, not the type the function is called on. If they are the same, `this.type` is the shortest solution.